### DISCUSSION TITLE
Separate items when joining

### DISCUSSION THREAD

**Discussion by boyanmolina** [hluk/CopyQ#2176] [Q&A] [answered]
When joining several text string in the Command module, how do I introduce an separator (empty line) between the texts I am joining?
Thanks

**Comment by hluk** [comment, marked as answer]
The following command merges selected items with a custom `separator` into new item ([here is how to add the command to CopyQ](http://copyq.readthedocs.io/en/latest/faq.html#how-to-load-shared-commands-and-share-them)):

```ini
[Command]
Command="
    copyq:
    const separator = '\\n---\\n';
    var sel = new ItemSelection().current();
    const texts = sel.itemsFormat(mimeText);
    sel.selectAll();
    add(texts.join(separator));
    sel.invert();
    selectItems(sel.rows()[0]);"
Icon=\xf0c1
InMenu=true
Name=Join
Shortcut=shift+space
```

**Edit:** I've added the command to the copyq-commands repository: https://github.com/hluk/copyq-commands/tree/master/Application#join-selected-items

**Reply by boyanmolina** [reply]
> The following command merges selected items with a custom `separator` into new item ([here is how to add the command to CopyQ](http://copyq.readthedocs.io/en/latest/faq.html#how-to-load-shared-commands-and-share-them)):
> 
> ```ini
> [Command]
> Command="
>     copyq:
>     const separator = '\\n---\\n';
>     var sel = new ItemSelection().current();
>     const texts = sel.itemsFormat(mimeText);
>     sel.selectAll();
>     add(texts.join(separator));
>     sel.invert();
>     selectItems(sel.rows()[0]);"
> Icon=\xf0c1
> InMenu=true
> Name=Join
> Shortcut=shift+space
> ```
> 
> **Edit:** I've added the command to the copyq-commands repository: https://github.com/hluk/copyq-commands/tree/master/Application#join-selected-items

Thank you, worked like a charm!
